crates/wash-cli/tests/wash_reg.rs

```diff
@@ -232,3 +246,291 @@ fn integration_reg_push_comprehensive() {
 
     remove_dir_all(push_dir).unwrap();
 }
+
+#[tokio::test]
+#[cfg_attr(
+    not(can_reach_wasmcloud_azurecr_io),
+    ignore = "wasmcloud.azurecr.io is not reachable"
+)]
+async fn intergration_reg_config() -> Result<()> {
```

Comment:
Thanks for adding to the test suite! These tests look great, but they're a bit focused on lower level bits -- I think they might be better as unit tests closer to the functions that you're testing.

Also, rather than pull a bunch of tests into one, let's split them out into little chunks -- testing precedence, with the field missing, etc can be their own test functions (in the configuration code, rather than here IMO), with their own `///` comments explaining what they do/what they test, if you're open to it.